### DISCUSSION TITLE
Subscribing to an event listener for auth errors with stateful components

### DISCUSSION THREAD

**Discussion by AlbertoPerez8** [Azure/communication-ui-library#3389] [Q&A]
Is there a way to integrate or subscribe to some of event listener compatible with the stateful components that is capable of notifying when the communications token is expired?

I know that it's possible to verify when a token is expired by validating the expiry date, but is there some other method of knowing when the token expires through some sort of error/event listener that catches the moment when it happens? 

I'd like to know if there's this alternative rather than having to set up a timed expiration check.

**Comment by mgamis-msft** [comment]
I am not aware of any event listener that fires when an ACS token expires. But if want to ensure your token is refreshed you can do this by creating a ACS token with `CommunicationTokenRefreshOptions`? Our Calling sample does this using this helper function in this [file](https://github.com/Azure/communication-ui-library/blob/main/samples/Calling/src/app/utils/credential.ts#L14).
